### ISSUE TITLE
The Beamtown Bullies lets you activate its ability on your turn

### ISSUE THREAD

**Issue by alexander-novo** [magefree/mage#9975]
[[The Beamtown Bullies]] lets you activate its ability on your turn, despite there being no valid targets. It asks you to target an opponent, won't let you target anything, and when when you hit cancel, it leaves the card tapped even though no ability was played.

**Comment by chesse20** [comment]
The part of targeting an opponent on their turn is before the : colon symbol. having a valid target is not a part of the cost of the ability and therefore you can activate it at any time even if you have no valid targets

**Comment by alexander-novo** [comment]
> The part of targeting an opponent on their turn is before the : colon symbol. having a valid target is not a part of the cost of the ability and therefore you can activate it at any time even if you have no valid targets

You can clearly see in the cardfetcher response above that is not the case. It is not possible to have the word "target" in a cost.

**Comment by chesse20** [comment]
yea i meant to say after

**Comment by PurpleCrowbar** [comment]
> The part of targeting an opponent on their turn is before the : colon symbol. having a valid target is not a part of the cost of the ability and therefore you can activate it at any time even if you have no valid targets

This is not true. Activated abilities with targets cannot be activated without a legal target. In the case of The Beamtown Bullies, it should not be activatable if there are no opponents whose turn it is (i.e., should not be activatable on its controller's turn)

**Comment by xenohedron** [comment]
The fundamental issue here is that the target is being added as part of the `apply()` method instead of on the card where it could take advantage of the standard targeting interface. The current implementation of `TargetOpponentWhoseTurnItIs` uses a `game` parameter which prevents that. If possible, I think should just refactor to `TargetPlayer(filter)` with appropriate predicates for opponent and active player. (If that can't work, then `TargetOpponentWhoseTurnItIs` would need to override several more methods.) Then adjust to use `EachTargetPointer`.
